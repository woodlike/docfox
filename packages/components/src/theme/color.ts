import { GenericThemeColor } from '.';

export const query = (
  colorQuery: string,
  color: GenericThemeColor,
  idx = 0,
): string | string[] => {
  const themeColor = color[colorQuery];
  if (Array.isArray(themeColor)) {
    return themeColor[idx];
  }
  return themeColor;
};
