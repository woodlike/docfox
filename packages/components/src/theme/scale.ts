export interface ScaledColorProps {
  color: string;
  idx: number;
}

export type Scale = string | number;
export type ScaleBox = [Scale, Scale, Scale, Scale];
export type ScaleArea = number | [number, number] | [number, number, number] | [number, number, number, number];

export function getThemeScale(area: ScaleArea, scale: Scale[]): Scale | Scale[] {
  return Array.isArray(area) ? area.map(value => scale[value]) : scale[area];
}

export function createBox(scale: Scale | Scale[]): ScaleBox {
  if (Array.isArray(scale)) {
    if (scale.length === 2) {
      return [...scale, ...scale] as ScaleBox;
    }
    if (scale.length === 3) {
      return [...scale, scale[1]] as ScaleBox;
    }
    return [...scale] as ScaleBox;
  }
  return [scale, scale, scale, scale];
}

export const create = (area: ScaleArea, scale: Scale[]): ScaleBox => createBox(getThemeScale(area, scale));

export const toCSSPixel = (box: ScaleBox): string => box.map(value => `${value}px`).join(' ');
export const toCSSString = (box: ScaleBox): string => box.map(value => `${value}`).join(' ');
