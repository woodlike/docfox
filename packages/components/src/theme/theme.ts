import { base } from '@theme-ui/presets';
import { theme as themeQuery, Colors, toRGB } from 'theme-query';

export interface Theme {
  readonly borderStyles: string[];
  readonly borderWidths: number[];
  readonly breakpoints: string[];
  readonly colors: ThemeColor & GenericThemeColor;
  readonly fonts: ThemeFontsProps;
  readonly fontSizes: number[];
  readonly space: number[];
}

export interface GenericThemeColor {
  readonly [key: string]: string | string[];
}

export interface ThemeFontsProps {
  readonly body: string;
  readonly heading: ThemeHeadingProps;
}

export interface ThemeHeadingProps {
  readonly display: string;
}

export interface ThemeColor {
  readonly primary: string;
  readonly secondary: string;
  readonly text: string;
  readonly border: string;
  readonly muted: string;
  readonly mutedHover: string;
  readonly headline: string;
  readonly link: string;
  readonly textInverted: string;
}

const borderStyles = [
  'none',
  'hidden',
  'dotted',
  'dashed',
  'solid',
  'double',
  'groove',
  'ridge',
  'inset',
  'outset',
];
const borderWidths = [1, 2, 3, 4];
const breakpoints = ['0px', '480px', '768px', '990px', '1024px', '1440px'];
const fontSizes: number[] = [12, 14, 16, 18, 20, 24, 32, 44, 72];
const fontWeights: number[] = [100, 200, 300, 400, 500, 600, 700, 800, 900];
const spaces = [0, 3, 6, 12, 18, 24, 30, 36, 48, 72];

const colors: Colors = {
  codeBg: toRGB([38, 42, 51]),
  corals: [toRGB([255, 113, 99]), toRGB([229, 85, 78])],
  whites: [
    toRGB([255, 255, 255]),
    toRGB([240, 246, 246]),
    toRGB([240, 240, 240]),
    toRGB([249, 246, 241]),
    toRGB([247, 247, 247]),
  ],
  grays: [
    toRGB([222, 223, 224]),
    toRGB([159, 188, 208]),
    toRGB([142, 149, 154]),
    toRGB([89, 103, 113]),
    toRGB([77, 77, 79]),
    toRGB([56, 72, 82]),
  ],
  blacks: [toRGB([34, 34, 34]), toRGB([51, 51, 51])],
  beiges: [toRGB([250, 245, 244])],
  sand: toRGB([204, 153, 102]),
  yellows: [toRGB([223, 199, 129]), toRGB([255, 202, 84])],
};
export const theme = {
  ...base,
  ...themeQuery,

  borderStyles,
  borderWidths,
  breakpoints,
  colors: {
    ...colors,
    background: colors.whites[0],
    border: colors.grays[0],
    headline: colors.grays[5],
    link: colors.grays[3],
    primary: colors.corals[0],
    muted: colors.grays[3],
    mutedHover: colors.grays[2],
    secondary: colors.corals[1],
    text: colors.blacks[0],
    textInverted: colors.whites[1],
  },
  fontSizes,
  fontWeights,
  fonts: {
    ...themeQuery.fonts,
    heading: {
      display: `"Museo", serif`,
      secondary: `"MuseoSans", Helvetica, sans-serif`,
      campaign: `"Challista_signature", serif`,
    },
    monospace: `"IBM Plex Mono", monospace`,
  },
  header: ['70px', '70px', '70px', '80px', '80px', '80px'],
  letterSpacings: [0.5, 1, 1.5],
  space: spaces,
  transition: {
    duration: [...themeQuery.transition.duration],
    timing: [
      ...themeQuery.transition.timing,
      'cubic-bezier(0.000, 0.755, 0.450, 0.910)',
      'cubic-bezier(0.770, 0.000, 0.175, 1.000)',
    ],
  },
};
