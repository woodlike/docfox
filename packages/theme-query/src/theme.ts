import {
  borderWidths,
  colors,
  fontFamily,
  fontSizes,
  fontWeights,
  letterSpacings,
  spaces,
  timing,
  duration,
} from './tokens';

export const theme = {
  borderWidths,
  colors,
  fonts: {
    ...fontFamily,
  },
  fontSizes,
  fontWeights,
  letterSpacings,
  spaces,
  transition: {
    duration,
    timing,
  },
};
export default theme;
