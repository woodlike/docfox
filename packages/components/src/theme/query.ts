import { create, ThemeQuery, ThemeQueryConfig } from 'theme-query';
import { useThemeUI } from 'theme-ui';
import { theme } from './theme';

export const qt = create({ theme, styles: 'object' });

export interface ThemeQueryHook {
  qt: ThemeQuery;
}

export function useThemeQuery(): ThemeQueryHook {
  const { theme: themeUI } = useThemeUI();
  const config: ThemeQueryConfig = {
    theme: themeUI ? themeUI : theme,
    styles: 'object',
  };
  return { qt: create(config) };
}
