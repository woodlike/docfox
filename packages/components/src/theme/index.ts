import * as ColorQuery from './color';
import * as InternalScale from './scale';

export * from './query';
export * from './scale';
export * from './theme';

export const Color = { ...ColorQuery };
export const Scale = { ...InternalScale };
