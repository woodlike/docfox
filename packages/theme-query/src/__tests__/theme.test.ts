import { theme } from '../theme';

describe('Theme Object', () => {
  it('should match one to one with the exported theme', () => {
    expect(theme).toMatchSnapshot();
  });
});
