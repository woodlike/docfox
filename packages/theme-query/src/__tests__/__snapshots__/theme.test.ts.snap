// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme Object should match one to one with the exported theme 1`] = `
Object {
  "borderWidths": Array [
    1,
    2,
    3,
    4,
  ],
  "colors": Object {
    "beiges": Array [
      "rgb(250, 245, 244)",
    ],
    "blacks": Array [
      "rgb(34, 34, 34)",
      "rgb(51, 51, 51)",
    ],
    "corals": Array [
      "rgb(255, 113, 99)",
      "rgb(229, 85, 78)",
    ],
    "grays": Array [
      "rgb(222, 223, 224)",
      "rgb(159, 188, 208)",
      "rgb(142, 149, 154)",
      "rgb(89, 103, 113)",
      "rgb(77, 77, 79)",
      "rgb(56, 72, 82)",
    ],
    "sand": "rgb(204, 153, 102)",
    "whites": Array [
      "rgb(255, 255, 255)",
      "rgb(240, 246, 246)",
      "rgb(240, 240, 240)",
      "rgb(249, 246, 241)",
      "rgb(247, 247, 247)",
    ],
    "yellows": Array [
      "rgb(223, 199, 129)",
      "rgb(255, 202, 84)",
    ],
  },
  "fontSizes": Array [
    12,
    14,
    16,
    18,
    20,
    24,
    32,
    44,
  ],
  "fontWeights": Array [
    100,
    200,
    300,
    400,
    500,
    600,
    700,
    800,
    900,
  ],
  "fonts": Object {
    "body": "\\"MuseoSans\\", Helvetica, sans-serif",
    "heading": Object {
      "display": "\\"Museo\\", serif",
      "secondary": "\\"Museo500\\", serif",
    },
  },
  "letterSpacings": Array [
    0.5,
    1,
    1.5,
  ],
  "spaces": Array [
    0,
    3,
    6,
    12,
    18,
    24,
    30,
    36,
    48,
    72,
  ],
  "transition": Object {
    "duration": Array [
      0.33,
      0.5,
      0.66,
      0.99,
    ],
    "timing": Array [
      "cubic-bezier(0.455, 0.030, 0.515, 0.955)",
      "cubic-bezier(0.075, 0.820, 0.165, 1.000)",
      "cubic-bezier(1.000, 0.000, 0.000, 1.000)",
    ],
  },
}
`;
